docs/jwst/user_documentation/logging.rst

```diff
@@ -35,61 +35,82 @@ logs for those messages.
-of the pipeline, the configuration file should be named "stpipe-log.cfg".
-This file must be in the same directory in which you run the pipeline in order
-for it to be used.
+By default, either the command line interface or the ``call`` function for
+Step or Pipeline classes will log messages at the INFO level to the ``stderr``
+stream.
 
-If instead the configuration should be active only when specified,
-you should name it something other than "stpipe-log.cfg"; this filename should be
-specified using either the ``--logcfg`` parameter to the command line ``strun`` or
-using the ``logcfg`` keyword to a .call() execution of either a Step or Pipeline
-instance.
+The name of a file in which to save log information, as well as the desired
+level of logging messages, can be specified in optional command line arguments
+provided to ``strun``, or can be directly configured via the `logging` module
+in Python code.
 
-If this file does not exist, the default logging mechanism is STDOUT,
-with a level of INFO. An example of the contents of the stpipe-log.cfg file is:
+From the Command Line
+---------------------
 
-::
+The available options for the command line are:
 
-    [*]
-    handler = file:pipeline.log
-    level = INFO
+.. code-block:: text
 
-If there's no ``stpipe-log.cfg`` file in the working directory, which specifies
-how to handle process log information, the default is to display log messages
-to stdout.
+  --verbose, -v         Turn on all logging messages
+  --log-level LOG_LEVEL
+                        Log level (DEBUG, INFO, WARNING, ERROR, CRITICAL). Ignored if 'verbose' is specified.
+  --log-file LOG_FILE   Full path to a file name to record log messages
+  --log-stream LOG_STREAM
+                        Log stream for terminal messages (stdout, stderr, or null).
 
 For example:
-::
+
+.. code-block:: shell
 
     $ strun calwebb_detector1 jw00017001001_01101_00001_nrca1_uncal.fits
-        --logcfg=pipeline-log.cfg
+        --log-level=INFO --log-file=pipeline.log --log-stream=stdout
 
-Or in an interactive python environment:
-::
+will log messages to the terminal at the INFO level in the ``stdout`` stream
+and also record them to a file called "pipeline.log" in the current working directory.
 
-    result = Detector1Pipeline.call("jw00017001001_01101_00001_nrca1_uncal.fits",
-                                    logcfg="pipeline-log.cfg")
+To turn off all logging instead:
 
-and the file ``pipeline-log.cfg`` contains:
-::
+.. code-block:: shell
+
+    $ strun calwebb_detector1 jw00017001001_01101_00001_nrca1_uncal.fits --log-stream=null
+
+CRDS messages may still display, since its logger is separately configured.
+
+In Python Code
+--------------
+
+In a Python environment, more complex configuration is possible. For example,
+to configure logging to print only the message at the INFO level and direct time-stamped
+messages to a file at the DEBUG level::
+
+    import logging
```

Comment:
Should this block have `.. code-block:: python` to be consistent with your other changes?

Reply:
Same comment to all other similar occurrences.